src/coord/src/coord.rs

```diff
@@ -491,7 +524,13 @@ where
                     {
                         messages.push(StartupMessage::UnknownSessionDatabase);
                     }
-                    self.catalog.create_temporary_schema(session.conn_id());
+                    if self
+                        .catalog
+                        .create_temporary_schema(session.conn_id())
+                        .is_err()
+                    {
+                        messages.push(StartupMessage::OidExhaustion)
```

Comment:
I think just sending the error back to the client will make all the right things happen, and save you from needing the new StartupMessage. I think you had it like that in the first version—was there some reason that didn’t work out?